runtime/callback.c

```diff
@@ -53,23 +78,38 @@ static void init_callback_code(void)
 
 CAMLexport value caml_callbackN_exn(value closure, int narg, value args[])
 {
-  int i;
+  CAMLparam1(closure);
+  CAMLxparamN(args, narg);
+  CAMLlocal1(cont);
   value res;
+  int i;
+  caml_domain_state* domain_state = Caml_state;
 
   CAMLassert(narg + 4 <= 256);
+  domain_state->current_stack->sp -= narg + 4;
```

Comment:
`Stack_threshold` was changed from 256 words to 16 words, right? I think we need to check for stack overflow here (or restrict callbacks to a maximum of 12 arguments...)